### PR TITLE
S9 unlikes

### PR DESCRIPTION
**Pull request by PapaPwn**
Still has the bug described in slack, but to be fixed l8rs.

### CODE REVIEW

File: frontend/src/store/questions/index.js [hunk 1]
```diff
@@ -60,8 +60,9 @@ const mutations = {
     state.questionObj.answers = [ans, ...ordered]
   },
   FOCUS_LIKES (state, { likes }) {
+    const oldLikes = state.questionObj.question.likes
     state.questionObj.question.likes = likes
-    // TODO - ALREADY UPVOTED
+    state.questionObj.question.userLiked += likes - oldLikes
```
Comment (NunoDasNeves) [inline frontend/src/store/questions/index.js]: I thought we got userLikes in the object anyway?
This method is prone to races with other users isn't it?

File: frontend/src/store/reviews/index.js [hunk 2]
```diff
@@ -43,8 +43,9 @@ const mutations = {
     state.reviewObj.replies = sortByHotness(replies.map(replyMapper))
   },
   FOCUS_LIKES (state, { likes }) {
+    const oldLikes = state.reviewObj.review.likes
     state.reviewObj.review.likes = likes
-    // TODO - ALREADY UPVOTED
+    state.reviewObj.review.userLiked += likes - oldLikes
```
Comment (NunoDasNeves) [inline frontend/src/store/reviews/index.js]: ^ see above

File: frontend/src/store/questions/index.js (hunk 1 above)
Comment (PapaPwn) [reply frontend/src/store/questions/index.js]: Doing a PUT request for likes, just returns the new "likes" value for the object. userLiked is not userLikes, it is what the current user has voted (-1, 0, +1). However it is prone to a race. Perhaps I'll just return the correct "userLiked" value with the "likes" value?

File: backend/src/controllers/question.js [hunk 3]
```diff
@@ -6,16 +7,18 @@ const errorHandler = require('./error')
...
 /* GET question data. */
-exports.getQuestion = function ({ params }, res) {
+exports.getQuestion = function ({ user, params }, res) {
+    const userID = user && user.id || ANONYMOUS
     const getQuestion = Promise.all([
         questionModel.getQuestion(params.id),
-        likesModel.getLikes({ type: 'question', id: params.id })
-    ]).then(([question, likes]) => {
+        likesModel.getLikes({ type: 'question', id: params.id }),
+        likesModel.getUserLiked({ type: 'question', id: params.id, userID })
```
Comment (DarkPurple141) [inline backend/src/controllers/question.js]: I feel like this is an unneeded additional call if we just got return list of likes in the `.getLikes` call. Then you can deduce the second part trivially.

File: backend/src/controllers/question.js [hunk 4]
```diff
@@ -24,20 +27,26 @@ exports.getQuestion = function ({ params }, res) {
...
         // Get the answers
         commentModel.getComments({ questionID: params.id }, query.p)
             .then(answers => Promise.all([
                 answers,
                 Promise.all(
-                    answers.map(answer => likesModel.getLikes({ type: 'answer', id: answer.id }))
+                    answers.map(answer => likesModel.getLikes(
+                        { type: 'answer', id: answer.id }))
+                ),
+                Promise.all(
```
Comment (DarkPurple141) [inline backend/src/controllers/question.js]: Can get rid of this if you do that.

File: backend/src/controllers/review.js [hunk 5]
```diff
@@ -6,16 +7,18 @@ const errorHandler = require('./error')
...
+exports.getReview = function ({ user, params }, res) {
+    const userID = user && user.id || ANONYMOUS
     const getReview = Promise.all([
         reviewModel.getReview(params.id),
-        likesModel.getLikes({ type: 'review', id: params.id })
-    ]).then(([review, likes]) => {
+        likesModel.getLikes({ type: 'review', id: params.id }),
+        likesModel.getUserLiked({ type: 'review', id: params.id, userID })
+    ]).then(([review, likes, userLiked]) => {
         return Promise.all([
```
Comment (DarkPurple141) [inline backend/src/controllers/review.js]: unnecssary Promise.all()

File: backend/src/controllers/review.js [hunk 6]
```diff
@@ -24,19 +27,26 @@ exports.getReview = function ({ params }, res) {
...
+        // Get the replies
         commentModel.getComments({ reviewID: params.id }, query.p)
             .then(replies => Promise.all([
                 replies,
                 Promise.all(
-                    replies.map(reply => likesModel.getLikes({ type: 'reply', id: reply.id }))
+                    replies.map(reply => likesModel.getLikes(
+                        { type: 'reply', id: reply.id }))
+                ),
+                Promise.all(
```
Comment (DarkPurple141) [inline backend/src/controllers/review.js]: Still feel like this is redundant.

File: frontend/src/utils/api/index.js [hunk 7]
```diff
@@ -21,11 +21,6 @@ async function responseCheck(res) {
 function request (path, { headers, method, data }) {
   // eventually add cors and auth headers
   const url = `${API_URL}${path}`
-
-  if (method === 'GET') {
```
Comment (DarkPurple141) [inline frontend/src/utils/api/index.js]: I still don't think sending the auth token with every req is good.

Comment (PapaPwn) [reply frontend/src/utils/api/index.js]: I agree, but for now I think its okay

File: backend/src/controllers/review.js (hunk 5 above)
Comment (PapaPwn) [reply backend/src/controllers/review.js]: Resolved in reputation PR

File: frontend/src/store/questions/index.js (hunk 1 above)
Comment (PapaPwn) [reply frontend/src/store/questions/index.js]: Will address if I have time though

**Pull request merged**